tensorflow_probability/python/distributions/empirical_test.py

```diff
@@ -0,0 +1,694 @@
+        [[[0, 0, 0, 1], [0, 0, 1, 1]],
+         [[1, 1, 1, 2], [0, 1, 2, 2]]]
+    ]
+    expected_entropys = [
+        entropy([1]),
+        entropy([0.4, 0.2, 0.4]),
+        [entropy([1]), entropy([0.5, 0.5]), entropy([0.5, 0.5])],
+        [[entropy([0.75, 0.25]), entropy([0.5, 0.5])],
+         [entropy([0.75, 0.25]), entropy([0.25, 0.25, 0.5])]]
+    ]
+
+    for sample, expected_entropy in zip(samples, expected_entropys):
+      input_ = tf.convert_to_tensor(sample, dtype=np.float64)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=input_.shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph)
+      self.assertAllClose(self.evaluate(dist.entropy()), expected_entropy)
+
+  def testSampleN(self):
+    for samples_shape in ([2], [2, 4], [4, 2, 4]):
+      input_ = random_samples(samples_shape)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=samples_shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph)
+      self.assertAllEqual(self.evaluate(tf.shape(dist.sample())),
+                          dist.batch_shape_tensor())
+
+      n = 1000
+      seed = 42
+      self.assertAllEqual(self.evaluate(dist.sample(n, seed)),
+                          self.evaluate(dist.sample(n, seed)))
+
+  def testMean(self):
+    samples = [
+        [0, 1, 1, 2],
+        [[0, 0, 0, 1], [0, 1, 1, 2]],
+        [[[0, 0], [1, 1]],
+         [[0, 1], [0, 2]]]
+    ]
+    expected_means = [
+        1.,
+        [0.25, 1],
+        [[0, 1], [0.5, 1]]
+    ]
+
+    for sample, expected_mean in zip(samples, expected_means):
+      input_ = tf.convert_to_tensor(sample, dtype=np.float32)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=input_.shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph)
+      self.assertAllClose(self.evaluate(dist.mean()), expected_mean)
+
+  def testMode(self):
+    samples = [
+        [0, 1, 1, 2],
+        [[0, 0, 1], [0, 1, 1], [2, 2, 2]],
+        [[[0, 0, 0, 0], [0, 0, 1, 1]],
+         [[1, 1, 1, 2], [0, 1, 2, 2]]]
+    ]
+    expected_modes = [
+        1,
+        [0, 1, 2],
+        [[0, 0], [1, 2]]
+    ]
+
+    for sample, expected_mode in zip(samples, expected_modes):
+      input_ = tf.convert_to_tensor(sample, dtype=np.int32)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=input_.shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph)
+      self.assertAllClose(self.evaluate(dist.mode()), expected_mode)
+
+  def testVarianceAndStd(self):
+    samples = [
+        [0, 1, 1, 2],
+        [[1, 3], [2, 6]]
+    ]
+    expected_variances = [
+        0.5,
+        [1., 4.],
+    ]
+
+    for sample, expected_variance in zip(samples, expected_variances):
+      input_ = tf.convert_to_tensor(sample, dtype=np.float32)
```

Comment:
if you want to simplify a bit, you could simply do `placeholder_with_default(input=np.float32(sample), ...)`